source/common/upstream/upstream_impl.cc

```diff
@@ -960,6 +960,36 @@ createOptions(const envoy::config::cluster::v3::Cluster& config,
       config.has_http2_protocol_options(), validation_visitor);
 }
 
+LBPolicyConfig::LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config) {
+  // TODO: Need to verify that lb type and the config for the policy are forced to match in
+  // a 1-1 relationship. If they are, change this to a switch case using config.lb_policy()
```

Comment:
Sorry for the back-and-forth here.
Can we use `config.lb_config_case()` to be the switch differentiator here?
It couples the correct `oneof` that determines the correct type.
I would leave the TODO, explaining that the lb type reflects the right config value.

Reply:
Might be a silly question but what cases are in lb_config_case? Do we have an example of this being used already?

Reply:
Done. Sorry for delay, was sick the last week